File: benchmark/comparison/results/blind_py/20.py
class Some:
    def __init__(self, v):
        self.v = v

class NoneOpt:
    pass

def unwrap_or(opt, default):
    if isinstance(opt, Some):
        return opt.v
    return default

print(unwrap_or(Some(42), 0))
print(unwrap_or(NoneOpt(), 0))
print(unwrap_or(Some(-1), 99))
print(unwrap_or(NoneOpt(), -1))

File: benchmark/comparison/results/blind_ts/20.ts
type Option<T> = { tag: "Some"; value: T } | { tag: "None" };

function Some<T>(value: T): Option<T> { return { tag: "Some", value }; }
function None<T>(): Option<T> { return { tag: "None" }; }

function unwrap_or<T>(opt: Option<T>, def: T): T {
  if (opt.tag === "Some") return opt.value;
  return def;
}

console.log(unwrap_or(Some(42), 0));
console.log(unwrap_or(None<number>(), 0));
console.log(unwrap_or(Some(-1), 99));
console.log(unwrap_or(None<number>(), -1));
